Packages/com.unity.inputsystem/InputSystem/Plugins/DualShock/DualShockGamepadHID.cs

```diff
@@ -292,6 +292,38 @@ public override void SetMotorSpeeds(float lowFrequency, float highFrequency)
             m_HighFrequenceyMotorSpeed = highFrequency;
         }
 
+        /// <summary>
+        /// Set motor speeds of both motors and the light bar color simultaneously.
+        /// </summary>
+        /// <param name="lowFrequency"><see cref="Haptics.IDualMotorRumble.SetMotorSpeeds"/></param>
+        /// <param name="highFrequency"><see cref="Haptics.IDualMotorRumble.SetMotorSpeeds"/></param>
+        /// <param name="color"><see cref="IDualShockHaptics.SetLightBarColor"/></param>
+        /// <returns>True if the command succeeded. Will return false if another command is currently being processed.</returns>
+        /// <remarks>
+        /// Use this method to set both the motor speeds and the light bar color in the same call. This method exists
+        /// because it is currently not possible to process an input/output control (IOCTL) command while another one
+        /// is in flight. For example, calling <see cref="SetMotorSpeeds"/> immediately after calling
+        /// <see cref="SetLightBarColor"/> might result in only the light bar color changing. The <see cref="SetMotorSpeeds"/>
+        /// call could fail. It is however possible to combine multiple IOCTL instructions into a single command, which
+        /// is what this method does.
+        ///
+        /// See <see cref="Haptics.IDualMotorRumble.SetMotorSpeeds"/> and <see cref="IDualShockHaptics.SetLightBarColor"/>
+        /// for the respective documentation regarding setting rumble and light bar color.</remarks>
+        public bool SetMotorSpeedsAndLightBarColor(float lowFrequency, float highFrequency, Color color)
+        {
+            var command = DualShockHIDOutputReport.Create();
+            command.SetMotorSpeeds(lowFrequency, highFrequency);
+            command.SetColor(color);
+
+            var result = ExecuteCommand(ref command);
```

Comment:
Looks good to me, but out of curiosity, will the execute command be queuing these commands when used like this (two actions on the same command) to avoid the problem with hw/interface not supporting both light color and motor speed being modulated at the same time? @andrew-oc

Reply:
@ekcoh No, device commands are ultimately just bit masks, so it's possible to combine multiple instructions into a single one. This fix just sends one command with the set light bar and rumble bits both set.